### ISSUE TITLE
May18 branch does not Build IL2CPP APPX

### ISSUE THREAD

**Issue by StephenHodgson** [Microsoft/MixedRealityToolkit-Unity#2125]
## Overview
When building the APPX package in Visual Studio the build fails to resolve external symbols for the Spectator View assemblies:

```
error LNK2001: unresolved external symbol _IsARKitSessionConfigurationSupported@0
error LNK2001: unresolved external symbol _IsARKitWorldTrackingSessionConfigurationSupported@0
error LNK2001: unresolved external symbol _unity_CreateNativeARSession@0
error LNK2001: unresolved external symbol _session_SetSessionCallbacks@24
error LNK2001: unresolved external symbol _session_SetPlaneAnchorCallbacks@16
error LNK2001: unresolved external symbol _session_SetUserAnchorCallbacks@16
error LNK2001: unresolved external symbol _StartWorldTrackingSession@20
error LNK2001: unresolved external symbol _StartWorldTrackingSessionWithOptions@24
error LNK2001: unresolved external symbol _StartSession@16
error LNK2001: unresolved external symbol _StartSessionWithOptions@20
error LNK2001: unresolved external symbol _PauseSession@4
error LNK2001: unresolved external symbol _HitTest@28
error LNK2001: unresolved external symbol _GetLastHitTestResult@4
error LNK2001: unresolved external symbol _GetVideoTextureHandles@0
error LNK2001: unresolved external symbol _GetAmbientIntensity@0
error LNK2001: unresolved external symbol _GetTrackingQuality@0
error LNK2001: unresolved external symbol _GetARPointCloud@8
error LNK2001: unresolved external symbol _SetCameraNearFar@8
error LNK2001: unresolved external symbol _CapturePixelData@12
error LNK2001: unresolved external symbol _SessionAddUserAnchor@72
error LNK2001: unresolved external symbol _SessionRemoveUserAnchor@8
```
Also two stack overflow warnings:
```
 warning C4717: 'ReplayKitRecorder_set_Controls_m1308905516': recursive on all control paths, function will cause runtime stack overflow
 warning C4717: 'ReplayKitRecorder_get_Controls_m576547881': recursive on all control paths, function will cause runtime stack overflow
```

## Unity Editor Version
2017.4.3f1 & 2018.1.0f2

## Mixed Reality Toolkit Release Version
a2871ff511690107422802f87a427def1b60666e

**Comment by StephenHodgson** [comment]
Only difference I could find was that it was expecting functions with an underscore (`_`) prefix but in the libraries they don't have the prefixes for each of the functions it failed on.

**Comment by timGerken** [comment]
@tobygilbert-ms

**Comment by TobyGilbert-ms** [comment]
Im working on a fix for this and I'll test it on a device on Monday and add a PR.

**Comment by davidkline-ms** [comment]
This was released in the 2017.4.0.0 release candidate
